File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2410.cpp
#include <iostream>
#include <map>

using namespace std;

int main(){
    map<int, bool> freq;
    int n, v, tot = 0;
    
    cin >> n;
    for(int i = 0; i < n; i++){
        cin >> v;
        if(!freq[v]){
            tot ++;
        }
        
        freq[v] = true;
    }
    
    cout << tot << endl;
    
    return 0;
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2410.py
n = int(input())

freq = {}
tot = 0

for i in range(10**6+1):
    freq[i] = False

for i in range(n):
    v = int(input())
    
    if(not freq[v]):
        tot += 1
    
    freq[v] = True
    
print(tot)